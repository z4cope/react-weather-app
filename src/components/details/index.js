import { useSelector } from 'react-redux';
import { Link } from 'react-router-dom';

import {
  Hero,
  State,
  Visibility,
  InfoWrapper,
  DetailsPageTitle,
  DetailsWrapper,
  DetailCard,
  DetailTitle,
  DetailNumber,
  NoData,
  NoDataText,
  BackButton,
} from './styles/details';

const Details = () => {
  const details = useSelector((state) => state.data.detailsArr);
  return (
    <>
      {details.name ? (
        <>
          <Hero background="./assets/maps.webp">
            <State>{details.name}</State>
            <Visibility>Time zone: {details.timezone}</Visibility>
          </Hero>
          <InfoWrapper>
            <DetailsPageTitle>Weather Details</DetailsPageTitle>
            <DetailsWrapper>
              <DetailCard>
                <DetailTitle>Clouds</DetailTitle>
                <DetailNumber>{details.clouds.all}</DetailNumber>
              </DetailCard>
              <DetailCard>
                <DetailTitle>Longnitude</DetailTitle>
                <DetailNumber>{details.coord.lon}</DetailNumber>
              </DetailCard>
              <DetailCard>
                <DetailTitle>Latitude</DetailTitle>
                <DetailNumber>{details.coord.lat}</DetailNumber>
              </DetailCard>
              <DetailCard>
                <DetailTitle>Temperature</DetailTitle>
                <DetailNumber>{details.main.temp}</DetailNumber>
              </DetailCard>
              <DetailCard>
                <DetailTitle>Visibility</DetailTitle>
                <DetailNumber>{details.visibility}</DetailNumber>
              </DetailCard>
              <DetailCard>
                <DetailTitle>Wind</DetailTitle>
                <DetailNumber>{details.wind.speed}</DetailNumber>
              </DetailCard>
            </DetailsWrapper>
          </InfoWrapper>
        </>
      ) : (
        <NoData>
          <NoDataText>No data</NoDataText>
          <Link to="/">
            <BackButton>Pick a country</BackButton>
          </Link>
        </NoData>
      )}
    </>
  );
};

export default Details;
