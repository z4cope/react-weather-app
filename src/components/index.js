export { default as Navbar } from './navbar';
export { default as Counties } from './counties';
