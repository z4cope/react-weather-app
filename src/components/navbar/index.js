import { useState } from 'react';
import { useLocation, Outlet, Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { getLocations } from '../../features/locations/locationSLice';

import {
  NavMenu,
  NavWrapper,
  Logo,
  BurgerMenu,
  BurgerSideMenu,
  ExitSideMenu,
  BackButton,
  ToolBar,
  Stats,
  Mic,
  Search,
  ActionMenu,
  CloseActionMenu,
  SearchContainer,
  StatsWrapper,
  StatsTitle,
  MicWrapper,
  MicTitle,
  SearchTitle,
  SearchBar,
  SubmitSearch,
} from './styles/header';

const Navbar = () => {
  const dispatch = useDispatch();
  const [actionMenu, setActionMenu] = useState('');
  const [activeMenu, setActiveMenu] = useState(false);
  const [activeSideMenu, setActiveSideMenu] = useState(false);
  const [locationSearch, setLocationSearch] = useState('London');
  const actionMenuHandle = (e) => {
    setActionMenu(e.target.id);
    setActiveMenu(activeMenu ? true : !activeMenu);
  };

  const pageLocation = useLocation().pathname;

  const searchHandler = (e) => {
    setLocationSearch(e.target.value);
  };

  const searchResult = (e) => {
    e.preventDefault();
    dispatch(getLocations(locationSearch));
    setLocationSearch('');
  };

  return (
    <>
      <NavMenu>
        <NavWrapper>
          {pageLocation !== '/' ? (
            <Link to="/">
              <BackButton background="./assets/back-button.png" />
            </Link>
          ) : (
            <BurgerMenu
              onClick={() => setActiveSideMenu(!activeSideMenu)}
              background="./assets/menu.png"
            />
          )}
          <BurgerSideMenu activeSideMenu={activeSideMenu}>
            {' '}
            <ExitSideMenu
              onClick={() => setActiveSideMenu(!activeSideMenu)}
              background="./assets/close.png"
            />{' '}
            Nothing in here yet!
          </BurgerSideMenu>

          <Logo>Weather App</Logo>
        </NavWrapper>
        <ToolBar>
          <Stats
            onClick={(e) => actionMenuHandle(e)}
            id="stats"
            background="./assets/phone.png"
          />
          <Mic
            onClick={(e) => actionMenuHandle(e)}
            id="mic"
            background="./assets/mic.png"
          />
          <Search
            onClick={(e) => actionMenuHandle(e)}
            id="search"
            background="./assets/search.png"
          />
        </ToolBar>
        <ActionMenu activeMenu={activeMenu}>
          <CloseActionMenu
            type="button"
            onClick={() => setActiveMenu(!activeMenu)}
            background="./assets/close.png"
          ></CloseActionMenu>
          {
            {
              stats: (
                <StatsWrapper>
                  {' '}
                  <StatsTitle>Stats</StatsTitle>{' '}
                </StatsWrapper>
              ),
              mic: (
                <MicWrapper>
                  {' '}
                  <MicTitle>Coming Soon</MicTitle>{' '}
                </MicWrapper>
              ),
              search: (
                <SearchContainer onSubmit={(e) => searchResult(e)}>
                  {' '}
                  <SearchTitle>Search Country</SearchTitle>
                  <SearchBar
                    value={locationSearch}
                    onChange={(e) => searchHandler(e)}
                    placeholder="ex. London"
                  />
                  <SubmitSearch background="./assets/right.png" type="submit" />
                </SearchContainer>
              ),
            }[actionMenu]
          }
        </ActionMenu>
      </NavMenu>
      <Outlet />
    </>
  );
};

export default Navbar;
