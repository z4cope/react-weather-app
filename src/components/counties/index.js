import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { v4 as uuidv4 } from 'uuid';
import { Link } from 'react-router-dom';
import { getLocations } from '../../features/locations/locationSLice';
import { getDetails } from '../../features/details/detailsSlice';

import {
  Hero,
  CountyTitle,
  StatesWrapper,
  Arrow,
  State,
  StateCard,
  Longitude,
} from './styles/counties';

const Counties = () => {
  const dispatch = useDispatch();
  const contries = [];
  let contry = '';
  useEffect(() => {
    dispatch(getLocations('Egypt'));
  }, [dispatch]);

  const locations = useSelector((state) => state.locations.locationsArr);

  const sendDetailsData = (lat, lon) => {
    dispatch(getDetails({ lat, lon }));
  };

  if (locations.length) {
    locations.map((loc) => {
      contries.push(loc);
      return loc;
    });

    contry = contries[0].name;
  }

  return (
    <>
      <Hero background="../assets/maps.webp">
        <CountyTitle>{contry}</CountyTitle>
      </Hero>
      <StatesWrapper>
        {locations.length ? (
          locations.map((loc) => (
            <Link
              onClick={() => sendDetailsData(loc.lat, loc.lon)}
              key={uuidv4()}
              to="/details"
            >
              <StateCard background="./assets/europe.png">
                <Arrow background="./assets/right.png" />
                <State>{loc.name}</State>
                <Longitude>{loc.lon}</Longitude>
              </StateCard>
            </Link>
          ))
        ) : (
          <>No locations yet</>
        )}
        <State></State>
      </StatesWrapper>
    </>
  );
};

export default Counties;
