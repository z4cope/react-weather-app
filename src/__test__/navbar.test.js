import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Navbar } from '../components';

jest.mock('react-redux');
describe('Testing for navbar component', () => {
  test('Snapshot for the navbar', () => {
    const tree = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
