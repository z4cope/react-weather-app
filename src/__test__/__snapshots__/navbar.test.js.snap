// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for navbar component Snapshot for the navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sc-bcXHqe jfQZQS"
      >
        <div
          class="sc-gswNZR dvtPDf"
        >
          <button
            class="sc-hLBbgP kzWCJy"
          />
          <div
            class="sc-eDvSVe cSWuy"
          >
             
            <button
              class="sc-jSUZER bZPjSO"
            />
             
            Nothing in here yet!
          </div>
          <h1
            class="sc-dkrFOg giGPfE"
          >
            Weather App
          </h1>
        </div>
        <div
          class="sc-ftTHYK jLmMgl"
        >
          <button
            class="sc-pyfCe fCtUSr"
            id="stats"
          />
          <button
            class="sc-jrcTuL jGeqSv"
            id="mic"
          />
          <button
            class="sc-kDvujY ktkeZP"
            id="search"
          />
        </div>
        <div
          class="sc-ipEyDJ elAuNy"
        >
          <button
            class="sc-csuSiG lglRZF"
            type="button"
          />
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-bcXHqe jfQZQS"
    >
      <div
        class="sc-gswNZR dvtPDf"
      >
        <button
          class="sc-hLBbgP kzWCJy"
        />
        <div
          class="sc-eDvSVe cSWuy"
        >
           
          <button
            class="sc-jSUZER bZPjSO"
          />
           
          Nothing in here yet!
        </div>
        <h1
          class="sc-dkrFOg giGPfE"
        >
          Weather App
        </h1>
      </div>
      <div
        class="sc-ftTHYK jLmMgl"
      >
        <button
          class="sc-pyfCe fCtUSr"
          id="stats"
        />
        <button
          class="sc-jrcTuL jGeqSv"
          id="mic"
        />
        <button
          class="sc-kDvujY ktkeZP"
          id="search"
        />
      </div>
      <div
        class="sc-ipEyDJ elAuNy"
      >
        <button
          class="sc-csuSiG lglRZF"
          type="button"
        />
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
